src/GraphQL.MicrosoftDI/SelfActivatingServiceProvider.cs

```diff
@@ -37,6 +37,19 @@ public SelfActivatingServiceProvider(IServiceProvider serviceProvider)
         }
 
         /// <inheritdoc cref="ActivatorUtilities.GetServiceOrCreateInstance(IServiceProvider, Type)"/>
-        public object GetService(Type serviceType) => ActivatorUtilities.GetServiceOrCreateInstance(_serviceProvider, serviceType);
+        public object GetService(Type serviceType)
+        {
+            // if the type is an interface, attempt to retrieve the interface registration from the
+            // underlying service provider or else return null. (May trigger an exception in a method
+            // calling GetRequiredService, of course.) But for concrete classes, attempt to
+            // create the instance if it is not able to be returned from the service provider.
+            // Note: this does not intelligently choose the constructor but rather tries the first one it encounters.
```

Comment:
Really? I think I was looking one time inside `GetServiceOrCreateInstance` and they choose.

Reply:
Well it does have code that you can use an attribute to mark which constructor to use. But besides that, I think it’s pretty dumb. It’s been a while since I looked at it, so going from memory here. I know MS DI will select the constructor based on what services have been registered. But GetServiceOrCreateInstance probably can’t do that because the only way to know if a service is registered is to request it. Just guessing of course.

Reply:
I'm fine either way.